### DISCUSSION TITLE
Data ingestion from local network

### DISCUSSION THREAD

**Discussion by hubmos** [dagster-io/dagster#25678] [Q&A]
Hi,
We run Dagster in a cloud VM, and now need to materialize an asset from a sql db inside our local network. What are the best practises for giving the cloud Dagster instance access inside our network?

In Azure Datafactory you can use a self hosted integration runtime, is there any equivalent in dagster?

**Comment by hubmos** [comment]
Is there no one that has been in the same situation? Right now im using SSH to accomplish the materializing, just wondering if theres a better way.

**Comment by garethbrickman** [comment]
I think the closest equivalent to [Azure Data Factory's self-hosted integration runtime](https://learn.microsoft.com/en-us/azure/data-factory/create-self-hosted-integration-runtime?tabs=data-factory) is [Dagster Pipes](https://docs.dagster.io/concepts/dagster-pipes).
